extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/authentication/endpoint/CoordinatorBasicAuthenticatorResourceHandler.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.security.basic.authentication.endpoint;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.smile.SmileFactory;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import com.google.inject.Inject;
+import io.druid.java.util.common.StringUtils;
+import io.druid.security.basic.BasicAuthUtils;
+import io.druid.security.basic.BasicSecurityDBResourceException;
+import io.druid.security.basic.authentication.BasicHTTPAuthenticator;
+import io.druid.security.basic.authentication.db.updater.BasicAuthenticatorMetadataStorageUpdater;
+import io.druid.security.basic.authentication.entity.BasicAuthenticatorCredentialUpdate;
+import io.druid.security.basic.authentication.entity.BasicAuthenticatorUser;
+import io.druid.server.security.Authenticator;
+import io.druid.server.security.AuthenticatorMapper;
+
+import javax.ws.rs.core.Response;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CoordinatorBasicAuthenticatorResourceHandler implements BasicAuthenticatorResourceHandler
+{
+  private final BasicAuthenticatorMetadataStorageUpdater storageUpdater;
+  private final Map<String, BasicHTTPAuthenticator> authenticatorMap;
+  private final ObjectMapper objectMapper;
+
+  @Inject
+  public CoordinatorBasicAuthenticatorResourceHandler(
+      BasicAuthenticatorMetadataStorageUpdater storageUpdater,
+      AuthenticatorMapper authenticatorMapper
+  )
+  {
+    this.storageUpdater = storageUpdater;
+    this.objectMapper = new ObjectMapper(new SmileFactory());
```

Comment:
Looks that objectMapper can be injected using ```@Smile``` annotation.

Reply:
Changed to use `@Smile` annotation